backend/models.py

```diff
@@ -0,0 +1,57 @@
+
+# Create your models here.
+from django.contrib.auth.models import AbstractUser
+from django.db import models
+
+import uuid
+
+
+class User(AbstractUser):
+    # Using: username, password, first_name, last_name, email inherited from Abstractuser
+
+
+class Post(models.Model):
+    postId = models.UUIDField(
+        primary_key=True, default=uuid.uuid4, editable=False, unique=True)
+    author = models.ForeignKey(User, on_delete=models.CASCADE)
+    title = models.CharField(max_length=50)
+    timestamp = models.DateTimeField(auto_now_add=True)
+    description = models.CharField(max_length=450)
+    # default posts are public
+    isPublic = models.BooleanField(default=True)
+
+
+class PostAccess(models.Model):
+    author = models.ForeignKey(User, on_delete=models.CASCADE)
+    post = models.ForeignKey(Post, on_delete=models.CASCADE)
+    accessId = models.ForeignKey(User, on_delete=models.CASCADE)
```

Comment:
Instead of managing access level like this, we can just have 2 fields
Sth like the following
```
visibility = models.CharField(max_length=10, choices={Public or Private or Friends or FoF, etc}, default="PUBLIC")
# contains users that are able to access the post
visible_to = ArrayField(models.CharField(max_length=500), blank=True, default=list)
```

Reply:
updated